ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -8722,8 +8722,18 @@ private ExprNodeDesc handleConversion(StructField tableField, ColumnInfo rowFiel
       // need to do some conversions here
       conversion.set(true);
       if (tableFieldTypeInfo.getCategory() != Category.PRIMITIVE) {
-        // cannot convert to complex types
-        column = null;
+        // handle array in case of complex types
+        String array_type_prefix = "array<";
```

Comment:
nit: For variable names, please use CamelCase.

Reply:
Sure, will update